### PR TITLE
Add `get_catalog_by_faculty` API endpoint

### PR DESCRIPTION
**Pull request by benny-n**

### CODE REVIEW

File: packages/server/src/api/students.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@ use std::{collections::HashMap, str::FromStr};
 
 use actix_web::{
     get, post, put,
-    web::{Data, Json, Query},
+    web::{Data, Json, Path, Query},
     HttpMessage, HttpRequest, HttpResponse,
 };
```
Comment (NissanOhana) [inline packages/server/src/api/students.rs:8]: we need one get API call for catalog:

GET catalog
no 'get' prefix.

if we dont have query param - get all.
if we have, sort.

Comment (NissanOhana) [reply packages/server/src/api/students.rs:8]: query param:
GET ..../sogrim/api/students/catalog/?faculty=nisso

File: packages/server/src/resources/catalog.rs [hunk 2]
```diff
@@ -12,11 +12,21 @@ use super::course::CourseId;
 
 pub(crate) type OptionalReplacements = Vec<CourseId>;
 
+#[derive(Default, Clone, Debug, Deserialize, Serialize)]
+pub enum Faculty {
+    #[default]
+    Unknown,
+    ComputerScience,
+    DataAndDecisionScience,
```
Comment (NissanOhana) [inline packages/server/src/resources/catalog.rs:20]: 😂

**Review by NissanOhana — APPROVED** [review]

**Comment by benny-n** [comment]
Migrations:

```js
db.Catalogs.updateMany({}, {$set: {"faculty": "Unknown"}})
db.Catalogs.updateMany({"name": {$regex: "נתונים"}}, {$set: {"faculty": "DataAndDecisionScience"}})
db.Catalogs.updateMany({"faculty": {$eq: "Unknown"}}, {$set: {"faculty": "ComputerScience"}})
db.Users.updateMany({"details.catalog": {$ne: null}}, {$set: {"details.catalog.faculty": "Unknown"}})
db.Users.updateMany({"details.catalog.name": {$regex: "נתונים"}}, {$set: {"details.catalog.faculty": "DataAndDecisionScience"}})
db.Users.updateMany({"details.catalog.faculty": {$eq: "Unknown"}}, {$set: {"details.catalog.faculty": "ComputerScience"}})
```

**Pull request merged**